versioned/storage/cache/src/main/java/org/projectnessie/versioned/storage/cache/CaffeineCacheBackend.java

```diff
@@ -49,8 +51,8 @@ static ImmutableCaffeineCacheBackend.Builder builder() {
   abstract MeterRegistry meterRegistry();
 
   @Value.Derived
-  Cache<CacheKey, byte[]> cache() {
-    Caffeine<CacheKey, byte[]> cacheBuilder =
+  Cache<CacheKeyValue, CacheKeyValue> cache() {
+    Caffeine<CacheKeyValue, CacheKeyValue> cacheBuilder =
         Caffeine.newBuilder().maximumWeight(capacity() * 1024L * 1024L).weigher(this::weigher);
```

Comment:
Have you considered using Caffeine's `expireAfter()` utility here? Stg like below seems doable:

```java
.expireAfter(
                new Expiry<>() {
                  @Override
                  public long expireAfterCreate(
                      CacheKeyValue key, CacheKeyValue value, long currentTime) {
                    return value.expiresAt == CACHE_UNLIMITED ? Long.MAX_VALUE : value.expiresAt;
                  }

                  @Override
                  public long expireAfterUpdate(
                      CacheKeyValue key,
                      CacheKeyValue value,
                      long currentTime,
                      long currentDuration) {
                    return expireAfterCreate(key, value, currentTime);
                  }

                  @Override
                  public long expireAfterRead(
                      CacheKeyValue key,
                      CacheKeyValue value,
                      long currentTime,
                      long currentDuration) {
                    return currentDuration;
                  }
                });
```